### ISSUE TITLE
`playwright/valid-title` doesn't enforce `mustMatch` in various cases

### ISSUE THREAD

**Issue by Firefds** [playwright-community/eslint-plugin-playwright#267]
## Preconditions
* Using latest Playwright - `1.42.0`
* Using latest `eslint-plugin-playwright` - `1.5.1`
* Have the following rule
   ```
   "playwright/valid-title": [
          "error",
          {
            "mustMatch": {
              "test": "@(local|all|nightly)$"
            }
          }
        ]
   ```
## Example of working passing lint
```test('title ends with @local', async () => {```

![image](https://github.com/playwright-community/eslint-plugin-playwright/assets/18083277/abb66557-e2fb-4b18-bfcd-75c1dd876918)

## Example of working failing lint
```test('title doesnt end with correct rule', async () => {```

![image](https://github.com/playwright-community/eslint-plugin-playwright/assets/18083277/b48a0310-457d-41a8-a0eb-d8c60eb86ac0)

## Cases where failing lint doesn't work (false positive)
1. Title includes template string
```test(`title with ${2} template string`, async () => {```

   ![image](https://github.com/playwright-community/eslint-plugin-playwright/assets/18083277/8661d5b1-cb5e-478c-b6f2-67d435eca05e)
2. Test includes new tags configuration
```test('title includes tag or annotation', {tag: '@tag'}, async () => {```

   ![image](https://github.com/playwright-community/eslint-plugin-playwright/assets/18083277/d0c77e63-8147-4a71-9622-9c4dfd823138)

**Comment by Firefds** [comment]
Closing. Needed to restart IDE :cry:
